Application/EdFi.Ods.Common/Validation/Validator.cs

```diff
@@ -0,0 +1,692 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using EdFi.Ods.Common.Extensions;
+using EdFi.Ods.Common.Models;
+using EdFi.Ods.Common.Models.Domain;
+using EdFi.Ods.Common.Models.Resource;
+
+namespace EdFi.Ods.Common.Validation
```

Comment:
This is copied from .NET 8 because modifications are needed to achieve the validation behavior we need (full rather than short-circuited validation, and incorporating Profiles restrictions).

Reply:
Do we need to remove this as part of the migration to .NET 8?

Reply:
This one needs to stay. It's a version of Validator.cs that is customized specifically for our needs with Profiles and to achieve the behavior that we're looking for for full object graph validation.